modules/ztp-nbde-troubleshooting-permanent-error-conditions.adoc

```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// scalability_and_performance/ztp-nbde-implementation-guide.adoc
+
+[id="ztp-nbde-troubleshooting-permanent-error-conditions_{context}"]
+= Troubleshooting permanent error conditions
+
+If the `READY` count does not equal the `DESIRED` count after an extended period of time, it might indicate a permanent failure condition. In this case, the following conditions might apply:
+
+* A typo in the Tang server URL or thumbprint in the `NEW_TANG_PIN` definition.
+* The Tang server is decommissioned or the keys are permanently lost.
+
+.Procedure
+
+. Validate the Tang server configuration by performing a simple encrypt and decrypt operation on each Tang server’s configuration as defined in the daemonset. For example:
++
+[source,terminal]
+----
+$ echo "okay" | clevis encrypt tang \
+  '{"url":"http://tangserver02:7500","thp":"badtumbprint"}' | \
+  clevis decrypt
+Unable to fetch advertisement: 'http://tangserver02:7500/adv/badthumbprint'!
+$ echo "okay" | clevis encrypt tang \
+  '{"url":"http://tangserver03:7500","thp":"goodthumbprint"}' | \
+  clevis decrypt
+okay
+----
+
+. When you have identified the root cause, remedy the underlying situation:
+
+.. Delete the non-working daemonset.
+.. Edit the daemonset definition to fix the underlying issue.  This might include any of the following actions:
++
+* Edit a Tang server entry to correct the URL and thumbprint.
+* Remove a Tang server that is no longer in service.
+* Add a new Tang server that is a replacement for a decommissioned server.
+
+. Distribute the updated daemonset again.
+
+[NOTE]
+====
+When replacing, removing, or adding a Tang server in configuration, the rekeying operation will succeed as long as at least one original server is still functional, including the server currently being rekeyed. If none of the original Tang servers are functional or can be recovered, recovery of the system is impossible and you must redeploy the affected nodes.
+====
+
+.Verification
+
+The logs of each daemonset-owned pod indicate whether the rekeying has completed successfully and might also indicate the failure condition. The following log is from a completed successful rekeying operation:
+
+[source,terminal]
+----
+$ oc logs rekey-tang-kp4q2
```

Comment:
This 1st line is the command, the rest of this block is the output.

Reply:
changed